### ISSUE TITLE
关于各个模型中的FM部分

### ISSUE THREAD

**Issue by DoubleYing** [zhougr1993/DeepInterestNetwork#16]
比如DeepFM：
`d_layer_fm_i = tf.concat([tf.reduce_sum(u_emb*i_emb, axis=-1, keep_dims=True), tf.gather(u_emb, [0], axis=-1) + tf.gather(i_emb, [0], axis=-1)], axis=-1)`
这里把u_emb和i_emb直接相乘作为二次交叉项，u_emb[0]和i_emb[0]作为一次项。
我是不是可以理解为：这里的field只有u和i两项，如果有更多field，是不是需要逐个进行二次交叉？
u_emb[0]作为一次项是不是损失了一些信息？用reduce_sum以后的u_emb会不会更好？

同理PNN：
`    din_i = tf.concat([u_emb, i_emb, u_emb*i_emb], axis=-1)`
PNN这里，对field进行二次交叉product，在有多个field时，是否也是需要逐个交叉？

**Comment by zhougr1993** [comment]
因为DeepFM Paper里的实现是这么描述的，是为了用随机的两维来实现FM的线性部分。然而根据我们的实验，DeepFM 在我们的数据集以及公开数据集上真正work的原因都是其中的PNN部分 也就是 被名为 Out product的 特征之间 element wise mult部分 `din_i = tf.concat([u_emb, i_emb, **u_emb*i_emb**], axis=-1)`
其FM部分没有什么效果。

**Comment by DoubleYing** [comment]
好的，谢谢您的回答
